webui/react/src/components/TableActionBar.tsx

```diff
@@ -148,54 +148,93 @@ const TableActionBar: React.FC<Props> = ({
     useModal(ExperimentTensorBoardModal);
   const isMobile = useMobile();
   const { openToast } = useToast();
-  const experimentIds = useMemo(
-    () => Array.from(selectedExperimentIds.keys()),
-    [selectedExperimentIds],
-  );
+  const validFilterSet = useObservable(formStore.validFilterSet);
 
-  const experimentMap = useMemo(() => {
-    return experiments.filter(Loadable.isLoaded).reduce(
-      (acc, experiment) => {
-        acc[experiment.data.experiment.id] = getProjectExperimentForExperimentItem(
-          experiment.data.experiment,
-          project,
-        );
+  const selectedProjectExperimentMap = useMemo(() => {
+    return [...selectedExperimentsMap].reduce(
+      (acc, [id, { experiment }]) => {
```

Comment:
and we can remove `as Record<number, ProjectExperiment>` in the bottom. 

```suggestion
      (acc: Record<number, ProjectExperiment>, [id, { experiment }]) => {
```